bundles/org.openhab.binding.goecharger/src/main/java/org/openhab/binding/goecharger/internal/handler/GoEChargerHandler.java

```diff
@@ -0,0 +1,333 @@
+                        count++;
+                    }
+                    if (goeResponse.energy[6] > 0) { // amps P3
+                        count++;
+                    }
+                    return new DecimalType(count);
+                case TEMPERATURE:
+                    return new QuantityType<>(goeResponse.temperature, SIUnits.CELSIUS);
+                case SESSION_CHARGE_CONSUMPTION:
+                    return new QuantityType<>((Double) (goeResponse.sessionChargeConsumption / 360000d),
+                            SmartHomeUnits.KILOWATT_HOUR);
+                case SESSION_CHARGE_CONSUMPTION_LIMIT:
+                    return new QuantityType<>((Double) (goeResponse.sessionChargeConsumptionLimit / 10d),
+                            SmartHomeUnits.KILOWATT_HOUR);
+                case TOTAL_CONSUMPTION:
+                    return new QuantityType<>((Double) (goeResponse.totalChargeConsumption / 10d),
+                            SmartHomeUnits.KILOWATT_HOUR);
+                case FIRMWARE:
+                    return new StringType(goeResponse.firmware);
+                case VOLTAGE_L1:
+                    return new QuantityType<>(goeResponse.energy[0], SmartHomeUnits.VOLT);
+                case VOLTAGE_L2:
+                    return new QuantityType<>(goeResponse.energy[1], SmartHomeUnits.VOLT);
+                case VOLTAGE_L3:
+                    return new QuantityType<>(goeResponse.energy[2], SmartHomeUnits.VOLT);
+                case CURRENT_L1:
+                    return new QuantityType<>((Double) (goeResponse.energy[4] / 10d), SmartHomeUnits.AMPERE);
+                case CURRENT_L2:
+                    return new QuantityType<>((Double) (goeResponse.energy[5] / 10d), SmartHomeUnits.AMPERE);
+                case CURRENT_L3:
+                    return new QuantityType<>((Double) (goeResponse.energy[6] / 10d), SmartHomeUnits.AMPERE);
+                case POWER_L1:
+                    return new QuantityType<>((Double) (goeResponse.energy[7] / 10d),
+                            SmartHomeUnits.WATT);
+                case POWER_L2:
+                    return new QuantityType<>((Double) (goeResponse.energy[8] / 10d),
+                            SmartHomeUnits.WATT);
+                case POWER_L3:
+                    return new QuantityType<>((Double) (goeResponse.energy[9] / 10d),
+                            SmartHomeUnits.WATT);
+            }
+        }
+        return UnDefType.UNDEF;
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
+            refresh();
+            return;
+        }
+
+        String key = null;
+        String value = null;
+        switch (channelUID.getId()) {
+            case MAX_AMPERE:
+                key = "amp";
+                value = ((QuantityType<?>)command).intValue() + "";
+                break;
```

Comment:
```suggestion
    @Override
    public void handleCommand(ChannelUID channelUID, Command command) {
        String key = null;
        String value = null;
        switch (channelUID.getId()) {
            case MAX_AMPERE:
                key = "amp";
                if (command instanceof DecimalType) {
                    value = String.valueOf(((DecimalType) command).intValue());
                } else if (command instanceof QuantityType<?>) {
                    value = String.valueOf((QuantityType<ElectricalCurrent>) command).toUnit(SmarthomeUnits.AMPERE).intValue());
                }
                break;
```

Something like this would be the correct way to do it. This also makes sure that QuantityType commands have the correct unit. Imagine someone sends a command  `10000 mA` (makes no sense, but you never now). You would probably send something like 10 for 10 amps instead of 10000 (which would happen if you take the intValue before conversion.

Reply:
Fine for me, I can do that later. Guess this makes it more stable.

Reply:
Thanks. I really like working with you.

Reply:
I just found how to define UoM in rules here:
https://www.openhab.org/docs/configuration/rules-dsl.html#number-item

So i wrote this code in a rule to test it:
```
var current = 5|A
var mA = current.toUnit("mA")
GoEChargerBindingThingMaxAmpere.sendCommand(mA)
```
This causes an error, however `GoEChargerBindingThingMaxAmpere.sendCommand(mA.intValue)` works, `GoEChargerBindingThingMaxAmpere.sendCommand("5000")` also works and is converted to DecimalType.

So to me it looks like it is not possible to send a QuantityType as command. Or am i doing anything wrong here?

Reply:
```
var current = 5|A
var mA = current.toUnit("mA")
GoEChargerBindingThingMaxAmpere.sendCommand(mA.toString)
```

Reply:
Yeah that one works. It's of type QuantityType. Will upload the changes in a sec.